### PR TITLE
Clean build: remove some compiler warnings

### PR DESCRIPTION
**Pull request by brianlheim**
Toward #2927.

- use correct enum for comparison between values in IDE
- remove assignment to an unused variable
- deactivate a good but unused debugging function via `#if 0`
- move a variable inside a preprocessor guard that would otherwise be unused

This eliminates 5 clang warnings.

### CODE REVIEW

**Comment by kisielk** [comment]
LGTM

**Comment by LFSaw** [comment]
awesome thanks!

File: common/SC_fftlib.cpp [hunk 1]
```diff
@@ -300,7 +300,9 @@ scfft * scfft_create(size_t fullsize, size_t winsize, SCFFT_WindowFunction winty
 void scfft_ensurewindow(unsigned short log2_fullsize, unsigned short log2_winsize, short wintype)
 {
 	// Ensure we have enough space to do our calcs
+#if SC_FFT_FFTW
```
Comment (vivid-synth) [inline common/SC_fftlib.cpp]: @brianlheim can you explain why we can remove this?

Comment (brianlheim) [reply common/SC_fftlib.cpp]: it's in the commit message - the variable is only used in code turned on by this preproc switch.

**Pull request merged**